src/Microsoft.Azure.WebJobs.Host/PropertyHelper.cs

```diff
@@ -1,230 +1,526 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
-using System.Diagnostics.Contracts;
+using System.Diagnostics;
 using System.Linq;
 using System.Reflection;
 
 namespace Microsoft.Azure.WebJobs.Host
 {
-    /// <summary>
-    /// Class used to facilitate reflection operations.
-    /// </summary>
+    // Original code here: https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs
```

Comment:
I had already pulled a copy into the Script repo and added set capabilities because they were lacking. Likely that version from script would work here as well. Your call. Benefit would be that we're using the same code, and perhaps could end up making it public in SDK so we can share (TBD).